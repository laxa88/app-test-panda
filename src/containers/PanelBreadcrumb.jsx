import React from "react";

const PanelBreadcrumb = props => <div>Breadcrumb panel</div>;

export default PanelBreadcrumb;
