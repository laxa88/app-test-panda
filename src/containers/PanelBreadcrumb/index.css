.body {
  height: 40px;
  display: flex;
  align-items: center;
  padding: 0 40px;
}

.homeIcon {
  font-size: 14px;
}

.arrow {
  opacity: 0.1;
  padding: 0 11px;
}

.link {
  font-size: 12px;
}
