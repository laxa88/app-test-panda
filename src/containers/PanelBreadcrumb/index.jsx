import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

import css from './index.css';
import Link from '../../components/Link';
import { ThemeContext } from '../../theme';

class PanelBreadcrumb extends React.PureComponent {
  render() {
    const { theme } = this.context;

    return (
      <div className={css.body}>
        <FontAwesomeIcon
          className={css.homeIcon}
          color={theme.primaryColor1}
          icon={['fas', 'home']}
        />

        <FontAwesomeIcon
          className={css.arrow}
          icon={['fas', 'chevron-right']}
        />

        <Link className={css.link} href="/">
          Page 1
        </Link>

        <FontAwesomeIcon
          className={css.arrow}
          icon={['fas', 'chevron-right']}
        />

        <Link className={css.link} href="/">
          Breadcrumb
        </Link>

        <FontAwesomeIcon
          className={css.arrow}
          icon={['fas', 'chevron-right']}
        />

        <Link className={css.link} href="/" disabled>
          Current page
        </Link>
      </div>
    );
  }
}

PanelBreadcrumb.contextType = ThemeContext;

export default PanelBreadcrumb;
