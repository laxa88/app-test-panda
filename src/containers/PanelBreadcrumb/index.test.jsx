import React from 'react';
import renderer from 'react-test-renderer';

import PanelBreadcrumb from './index';

describe('<PanelBreadcrumb />', () => {
  it('renders', () => {
    const result = renderer.create(<PanelBreadcrumb />);
    expect(result.toJSON()).toMatchSnapshot();
  });
});
