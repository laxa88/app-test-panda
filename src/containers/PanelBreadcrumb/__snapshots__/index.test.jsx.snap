// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelBreadcrumb /> renders 1`] = `
<div
  className="body"
>
  Mocked_FontAwesomeIcon (fas,home)
  Mocked_FontAwesomeIcon (fas,chevron-right)
  <a
    className="link sc-bwzfXH kxbnci"
    href="/"
  >
    Page 1
  </a>
  Mocked_FontAwesomeIcon (fas,chevron-right)
  <a
    className="link sc-bwzfXH kxbnci"
    href="/"
  >
    Breadcrumb
  </a>
  Mocked_FontAwesomeIcon (fas,chevron-right)
  <div
    className="link sc-bdVaJa hcEiKe"
    href="/"
  >
    Current page
  </div>
</div>
`;
